src/main/java/com/google/sps/data/FakeScheduledInterviewDao.java

```diff
@@ -0,0 +1,128 @@
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.Set;
+
+/** Mimics accessing Datastore to support managing ScheduledInterview entities. */
+public class FakeScheduledInterviewDao implements ScheduledInterviewDao {
+  // data is the fake database
+  public Map<Long, ScheduledInterview> data;
+
+  /** Initializes the fields for ScheduledInterviewDatastoreDAO. */
+  public FakeScheduledInterviewDao() {
+    data = new LinkedHashMap<Long, ScheduledInterview>();
+  }
+
+  /**
+   * Retrieves a scheduledInterviewEntity from storage and returns it as a ScheduledInterview
+   * object.
+   */
+  @Override
+  public Optional<ScheduledInterview> get(long id) {
+    if (data.containsKey(id)) {
+      return Optional.of(data.get(id));
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Retrieves all scheduledInterview entities from storage that involve a particular user and
+   * returns them as a list of ScheduledInterview objects in the order in which they occur.
+   */
+  @Override
+  public List<ScheduledInterview> getForPerson(String email) {
+    List<ScheduledInterview> relevantInterviews = new ArrayList<>();
+    List<ScheduledInterview> scheduledInterviews = new ArrayList<ScheduledInterview>(data.values());
+    scheduledInterviews.sort(
+        (ScheduledInterview s1, ScheduledInterview s2) -> {
+          if (s1.when().start().equals(s2.when().start())) {
+            return 0;
+          }
+          if (s1.when().start().isBefore(s2.when().start())) {
+            return -1;
+          }
+          return 1;
+        });
+
+    for (ScheduledInterview scheduledInterview : scheduledInterviews) {
+      if (email.equals(scheduledInterview.interviewerEmail())
+          || email.equals(scheduledInterview.intervieweeEmail())) {
+        relevantInterviews.add(scheduledInterview);
+      }
+    }
+    return relevantInterviews;
+  }
+
+  /** Returns a list of all scheduledInterviews ranging from minTime to maxTime of a user. */
```

Comment:
Please update this comment to indicate whether the returned interviews are in order or not (it is somewhat ambiguous from the current wording). Feel free to update the comment on the non-fake class to match.

Reply:
Done